vms/proposervm/vm.go

```diff
@@ -298,15 +301,24 @@ func (vm *VM) GetBlock(ctx context.Context, id ids.ID) (snowman.Block, error) {
 	return vm.getBlock(ctx, id)
 }
 
+func (vm *VM) GetPreference() ids.ID {
+	return vm.preferred
+}
+
 func (vm *VM) SetPreference(ctx context.Context, preferred ids.ID) error {
-	if vm.preferred == preferred {
-		return nil
-	}
```

Comment:
Originally I removed this because the engine calls `GetPreference` to pass the result into `SetPreference` on the proposervm. If the VM already sets `preference` in initialize, this means that the rest of the `SetPreference` code won't run w/ this check. Maybe a workaround could be to have a separate `initialPreference` field in the vm struct which is only used for the return value of `GetPreference` since it's only called once.... but that seems pretty weird too.... will see if I can think of a better way to do this.

Reply:
I think we can alternatively just move the first `SetPreference` call to the proposervm

Reply:
As discussed offline, I think it's better to get rid of `initialPreference` and call `SetPreference` on the innerVM from the ProposerVM's `Initialize` function.

This makes two changes (1) we call `SetPreference` from the VM itself where before we'd only call it from the engine and (2) this calls `SetPreference` before we start normal execution of the consensus engine ie. prior to completing bootstrapping. We should confirm this doesn't cause any issues, but this is not a documented invariant and I don't think any of our VMs make any assumptions about this.